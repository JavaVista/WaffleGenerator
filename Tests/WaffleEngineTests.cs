using System;
using System.Diagnostics;
using System.Text;
using ApprovalTests;
using WaffleGenerator;
using Xunit;
using Xunit.Abstractions;

public class WaffleEngineTests
{
    ITestOutputHelper output;

    public WaffleEngineTests(ITestOutputHelper output)
    {
        this.output = output;
    }

    [Fact]
    public void TextWaffleSample()
    {
        var waffleEngine = new WaffleEngine();
        var builder = new StringBuilder();
        waffleEngine.TextWaffle(
            paragraphs: 1,
            includeHeading: true,
            builder);
        Debug.WriteLine(builder.ToString());
    }

    [Fact]
    public void HtmlWaffleSample()
    {
        var waffleEngine = new WaffleEngine();
        var builder = new StringBuilder();
        waffleEngine.HtmlWaffle(
            paragraphs: 1,
            includeHeading: true,
            builder);
        Debug.WriteLine(builder.ToString());
    }

    [Fact]
    public void TextWaffleSingle()
    {
        var random = new Random(0);
        var waffleEngine = new WaffleEngine(random);
        var builder = new StringBuilder();
        waffleEngine.TextWaffle(1, true, builder);
        Approvals.Verify(builder.ToString());
    }

    [Fact]
    public void HtmlWaffleSingle()
    {
        var random = new Random(0);
        var waffleEngine = new WaffleEngine(random);
        var builder = new StringBuilder();
        waffleEngine.HtmlWaffle(1, true, builder);
        Approvals.Verify(builder.ToString());
    }

    [Fact]
    public void TextWaffleMultiple()
    {
        var random = new Random(0);
        var waffleEngine = new WaffleEngine(random);
        var builder = new StringBuilder();
        waffleEngine.TextWaffle(11, true, builder);
        Approvals.Verify(builder.ToString());
    }

    [Fact]
    public void HtmlWaffleMultiple()
    {
        var random = new Random(0);
        var waffleEngine = new WaffleEngine(random);
        var builder = new StringBuilder();
        waffleEngine.HtmlWaffle(11, true, builder);
        Approvals.Verify(builder.ToString());
    }

    [Fact]
    public void TextWaffleNoHeading()
    {
        var random = new Random(0);
        var waffleEngine = new WaffleEngine(random);
        var builder = new StringBuilder();
        waffleEngine.TextWaffle(1, false, builder);
        Approvals.Verify(builder.ToString());
    }

    [Fact]
    public void HtmlWaffleNoHeading()
    {
        var random = new Random(0);
        var waffleEngine = new WaffleEngine(random);
        var builder = new StringBuilder();
        waffleEngine.HtmlWaffle(1, true, builder);
        Approvals.Verify(builder.ToString());
    }
}