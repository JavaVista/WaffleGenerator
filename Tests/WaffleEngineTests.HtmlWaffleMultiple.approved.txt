<html>
<head>
<title>The Aesthetic Of Economico-Social Disposition</title>
</head>
<body>
<h1>The Aesthetic Of Economico-Social Disposition</h1>
<blockquote>"In this regard, the underlying surrealism of the  take home message should not divert attention from The Aesthetic Of Economico-Social Disposition"<br><cite>Humphrey Yokomoto in The Journal of the Total Entative Item (20044U)</cite></blockquote>
<h2>causation of milieu.</h2>
<p>On any rational basis, a particular factor, such as the functional baseline, the analogy of object, the strategic requirements or the principal overriding programming provides an interesting insight into the complementary functional derivation. This trend may dissipate due to the mensurable proficiency.  </p>
<p>Strictly speaking, the dangers inherent in the internal resource capability is constantly directing the course of the key consistent milieu. One must therefore dedicate resources to the reproducible geometric conventionalism immediately..  </p>
<p>On any rational basis, any solution to the problem of any element of volatility should touch base with the functional extrinsic milieu. This may be due to a lack of a responsive consistent transformation..  
<h2>The Final Consolidation.</h2>
<p>Clearly, it is becoming possible to resolve the difficulties in assuming that the basis of the  big picture identifies the probability of project success and the strategic overriding competence. We need to be able to rationalize The major theme of the optical reproducible reconstruction. The advent of the set of constraints rigorously posits the work being done at the 'coal-face'.  </p>
<p>
<h2>The Set Of Constraints.</h2>
<p>Few would deny that a realization the importance of the purchaser - provider cannot compare in its potential exigencies with The total quality objectives.  It might seem reasonable to think of a unique facet of prevalent arbitrary contingency as involving any comprehensive comprehensive component. Nevertheless, what amounts to the purchaser - provider enhances the importance of other systems and the necessity for the objective baseline. This trend may dissipate due to the consultative consistent baseline.  It might seem reasonable to think of the areas of particular expertise as involving the criterion of compatible dynamic verification. Nevertheless, the target population for the feasibility of the logical subjective milieu uniquely represents the access to corporate systems and what is beginning to be termed the "transitional theoretical reciprocity".  In respect to specific goals, the assertion of the importance of the inductive resonant monologism could go the extra mile for any commonality between the integrated auxiliary antithesis and the crucial test option.  Within normal variability, the quest for the gap analysis exceeds the functionality of the responsive central change. This trend may dissipate due to the proactive sanctioned item.  The position in regard to the critical phylogenetic impulse is that an overall understanding of any consideration of the pivotal resonant element gives a win-win situation for this hypothetical cardinal substructure. This should present few practical problems.  As regards an implementation strategy for optical relative concept, We should put this one to bed. On the other hand, the incorporation of the fully interactive free-floating expression must intrinsically determine this general milestones. This should present few practical problems.  </p>
<p></p>
</body>
</html>
